lib/discourse.php

```diff
@@ -199,11 +199,9 @@ function sso_parse_request( $wp )
 
   function get_avatar_url( $user_id ) {
     $avatar = get_avatar( $user_id );
-    $doc = new DOMDocument();
-    $doc->loadHTML( $avatar );
-    $xpath = new DOMXPath( $doc );
-    $src = $xpath->evaluate( 'string(//img/@src)' );
-    return $src;
+    if( preg_match( "/src=['\"](.*?)['\"]/i", $avatar, $matches ) )
+      return utf8_uri_encode( $matches[1] );;
+    return NULL; 
```

Comment:
No need for `return NULL;` since PHP handles that on its own.

There's also 2 semicolons on the line above